### PR TITLE
Ting Wang Completed Client-Testing-Mini

### PR DESCRIPTION
**Pull request by nunulong**

### CODE REVIEW

File: tests/functions.test.js [hunk 1]
```diff
@@ -8,13 +8,14 @@ chai.use(sinonChai);
 
 describe('functions', () => {
   let testArray = [];
-  beforeEach = () => {
+  const each = cases.each;
```
Comment (lefrenk) [inline tests/functions.test.js]: Pro tip: you can use destructuring here:

`const { each } = cases;`

File: tests/functions.test.js [hunk 2]
```diff
@@ -37,15 +41,36 @@ describe('functions', () => {
...
+  describe('each', () => {
+
+    it('should be a function', () => {
+      expect(each).to.be.an('function');
+    });
+    it('should call a callback for each item in the array', () => {
+      const myLength = testArray.length;
+      const callBack = sinon.spy();
+      each(testArray, callBack);
+      expect(callBack).to.have.callCount(7);
```
Comment (lefrenk) [inline tests/functions.test.js]: Excellent! 🤘

File: tests/functions.test.js [hunk 3]
```diff
@@ -37,15 +41,36 @@ describe('functions', () => {
...
+    });
+  });
+
+  describe('each', () => {
+
+    it('should be a function', () => {
+      expect(each).to.be.an('function');
+    });
+    it('should call a callback for each item in the array', () => {
+      const myLength = testArray.length;
```
Comment (lefrenk) [inline tests/functions.test.js]: Unused variable `myLength`

File: tests/types.test.js [hunk 4]
```diff
@@ -9,49 +9,79 @@ describe('Cases from app.js', () => {
...
   describe('Ben', () => {
     it('should be an object', () => {
       const Ben = cases.Ben;
-      expect(Ben).to.be.an('array');
+      expect(Ben).to.be.an('object');
     });
     // write a test to check if Ben's favorite band is 'Beastie Boys'.
+    it('should have favorite band Beastie Boys', () => {
+      const Ben = cases.Ben;
+      expect(Ben).to.have.property('favBand').to.equal('Beastie Boys');
```
Comment (lefrenk) [inline tests/types.test.js]: Nicely done here! 👌

**Comment by lefrenk** [comment]
Solid work, overall! 👌

- All tests pass ✔️
- Code is consistent with JS idioms and style ✔️
- Clearly has a good grasp of the material ✔️

Not much else to say. Nice work!

**Comment by lefrenk** [comment]
Please close your pull request when you get a chance. Thanks!

File: tests/functions.test.js (hunk 1 above)
Comment (nunulong) [reply tests/functions.test.js]: Thanks so much for your suggestions. I will modify the codes as instructed.

File: tests/functions.test.js (hunk 2 above)
Comment (nunulong) [reply tests/functions.test.js]: :smile:

File: tests/functions.test.js (hunk 3 above)
Comment (nunulong) [reply tests/functions.test.js]: 👍

File: tests/types.test.js (hunk 4 above)
Comment (nunulong) [reply tests/types.test.js]: :bowtie:

**Pull request closed without merging**